website/addons/owncloud/static/owncloudNodeConfig.js

```diff
@@ -0,0 +1,128 @@
+        // Designated host, specified from select or input element
+        self.host = ko.pureComputed(function() {
+            return self.useCustomHost() ? self.customHost() : self.selectedHost();
+        });
+        // Hosts visible in select element. Includes presets and "Other" option
+        self.visibleHosts = ko.pureComputed(function() {
+            return self.hosts().concat([otherString]);
+        });
+        // Whether to use select element or input element for host designation
+        self.useCustomHost = ko.pureComputed(function() {
+            return self.selectedHost() === otherString;
+        });
+        self.credentialsChanged = ko.pureComputed(function() {
+            return self.nodeHasAuth() && !self.validCredentials();
+        });
+        self.showCredentialInput = ko.pureComputed(function() {
+            return (self.credentialsChanged() && self.userIsOwner()) ||
+                (!self.userHasAuth() && !self.nodeHasAuth() && self.loadedSettings());
+        });
+
+        // Overrides
+        var defaults = {
+            onPickFolder: function(evt, item) {
+                evt.preventDefault();
+                var name = item.data.path !== '/' ? item.data.path : '/ (Full ' + self.addonName + ')';
+                var folder = {
+                    name: name,
+                    path: item.data.path,
+                    id: item.data.id
+                };
+                self.selected(folder);
+                self.folder(folder);
+                return false; // Prevent event propagation
+            },
+            connectAccount: function() {
+                window.location.href = this.urls().auth;
+            },
+            decodeFolder: function(folder_name) {
+                return folder_name;
+            }
+        };
+        self.options = $.extend({}, defaults, opts);
+        // Treebeard config
+        self.treebeardOptions = $.extend(
+            {},
+            OauthAddonFolderPicker.prototype.treebeardOptions,
+            {
+                onPickFolder: function(evt, item) {
+                    return this.options.onPickFolder.call(this, evt, item);
+                }.bind(this),
+                resolveLazyloadUrl: function(item) {
+                    return item.data.urls.folders;
+                },
+                decodeFolder: function(item) {
+                    return this.options.decodeFolder.call(this, item);
+                }.bind(this)
+            }
+        );
+
+    },
+    clearModal : function() {
+        var self = this;
+        self.message('');
+        self.messageClass('text-info');
+        self.selectedHost(null);
+        self.customHost(null);
+    },
+    connectAccount : function() {
+        var self = this;
+        if( !self.selectedHost() ){
+            self.changeMessage("Please select a OwnCloud repository.", 'text-danger');
+            return;
+        }
+        var url = self.urls().create;
```

Comment:
`url` doesn't get defined here, probably because it's not being returned by the serializer:

<img width="616" alt="screen shot 2016-09-13 at 10 39 02" src="https://cloud.githubusercontent.com/assets/5659262/18478233/7dc5bfb6-799e-11e6-835d-70b34e5dbef4.png">